tests/unit/introspection.tcl

```diff
@@ -27,8 +27,48 @@ start_server {tags {"introspection"}} {
         $rd monitor
         $rd read ;# Discard the OK
         r eval {redis.call('set',KEYS[1],ARGV[1])} 1 foo bar
-        assert_match {*eval*} [$rd read]
         assert_match {*lua*"set"*"foo"*"bar"*} [$rd read]
+        assert_match {*eval*} [$rd read]
```

Comment:
in the old code, there was a special hack that made sure EXEC is sent after the transaction instead of before.
now this happens implicitly, so i think we can afford to add a similar hack that makes sure EVAL is sent to the monitor BEFORE (as it used to be)

Reply:
Sure, I didn't really like the hack and wanted to skip it. I'm fine redoing the eval statement.